### DISCUSSION TITLE
Problems installing on a fresh Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by MantraMediaMike** [open-meteo/open-meteo#412] [Q&A] [answered]
Hi,

I have a fresh Ubuntu 22.04 installation and follow the instructions at https://github.com/open-meteo/open-meteo/blob/main/docs/getting-started.md but I run into these problems:

ERA5:

- The cdsapi library is missing. I installed it then with `pip install cdsapi` and it seems to work as I am not getting an error message anymore.

- After successful installation of the cdsapi I am getting a 401 running eg `openmeteo-api download-era5 era5_land -t 20230101-20230131 --cdskey 123456:7890-12345-..` but calling the endpoint directly with the credentials using CURL actually works. I also tried putting the credentials into $HOME/.cdsapi as mentioned on the era5 api but this also fails

Digital elevation model:

- `sudo apt install gdal` does not work as there is no gdal distribution for 22.04. I then followed the instructions at https://mothergeo-py.readthedocs.io/en/latest/development/how-to/gdal-ubuntu-pkg.html#putting-it-all-together and that seems to do the trick

Now I am just confused as I am not sure if I do something wrong, especially with the cdsapi, or if I am missing something?

**Comment by patrick-zippenfenig** [comment]
Hi, `cdsapi` is/was included in the Docker prebuilt-binaries. Ubuntu instructions could be improved. I am planing to replace the python CDS requirement by a native implementation.

For the `401` error, did you accept the ERA5-Land TOS in the CDS?

**Comment by MantraMedia** [comment]
Thank you for the fast reply and the explanation regarding cdsapi.

Yes, TOS were accepted


A direct curl request with the same credentials works as expected:

`curl -XPOST 'https://cds.climate.copernicus.eu/api/v2/resources/reanalysis-era5-land' -d '{"day":"01","time":["00:00","..."'`

and the response:

`{"state":"queued","request_id":"12345-67890-...","specific_metadata_json":{"top_request_origin":"api"}}`

**Reply by patrick-zippenfenig** [reply, marked as answer]
Did you format the cds key correctly? `--cdskey 123456:7890-12345-.` is a combination of userid and password

**Comment by MantraMedia** [comment]
Yes but I also found the bug now copying and pasting again. I just missed the last number of the key for whatever reason.

Thank you for patience and help!
